### PR TITLE
Add patient identifier endpoint

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR adds an endpoint for adding hashes to patient resources in a FHIR bundle. It also introduces a `utils` module for functionality that is shared across multiple endpoints.

### CODE REVIEW

File: containers/phdi-ingestion/app/routers/fhir_linkage_link.py [hunk 1]
```diff
@@ -1,13 +1,50 @@
...
+
+    input = dict(input)
+
+    if input.get("salt_str") in [None, ""]:
+        check_result = check_for_environment_variables(["SALT_STR"])
+        if check_result["status_code"] == 500:
+            response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+            return check_result
+        else:
+            input["salt_str"] = os.environ.get("SALT_STR")
```
Comment (nickclyde) [inline containers/phdi-ingestion/app/routers/fhir_linkage_link.py:48]: Nice, I like the flexibility to provide it as an environment variable here.

File: containers/phdi-ingestion/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+def test_check_for_fhir():
+    good_fhir = {"resourceType": "Patient"}
+    assert check_for_fhir(good_fhir) == good_fhir
+
+    bad_fhir = {}
+    with pytest.raises(AssertionError):
+        check_for_fhir(bad_fhir)
+
+
+def test_check_for_fhir_bundle():
```
Comment (nickclyde) [inline containers/phdi-ingestion/tests/test_utils.py:46]: Thanks for testing the utils!

**Review by nickclyde — APPROVED** [review]
Great work Dan!!

**Review by BradySkylight — APPROVED** [review]
This looks good to me though some of the python class definitions are new to me.  Great job!

**Pull request merged**